postoffice_django/publishing.py

```diff
@@ -85,21 +85,28 @@ class BulkPublisher(Publisher):
     def __init__(self, topic, payload, **attributes):
         super().__init__(topic=topic, payload=payload, bulk=True, **attributes)
 
-    def _create_message(self) -> list:
-        return [{
+    def _create_message(self) -> dict:
+        return {
             'topic': self.topic,
-            'payload': message,
+            'payload': self.payload,
             'attributes': self._stringify_attributes()
-        } for message in self.payload]
+        }
 
     def _create_publishing_error(self, error: str) -> None:
         PublishingError.objects.create(
             topic=self.topic,
-            payload=self.message,
+            payload=self._create_failed_message(),
             attributes=self.attributes,
             bulk=self.bulk,
             error=error,
         )
 
+    def _create_failed_message(self) -> list:
+        return [{
+           'topic': self.topic,
+           'attributes': self.attributes,
+           'payload': message_payload
+        } for message_payload in self.payload]
```

Comment:
Should a test be failing now? Because we used to save only one record, and now there is one per message in the payload.

Reply:
This `_create_failed_message()` does not save one record per message. It just generates a list of dicts, which will be saved as the payload of the unique record we generate